kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -307,6 +307,31 @@ private static List<Tuple2<List<String>, StructField>> recurseIntoComplexTypes(
     return filtered;
   }
 
+  /* Compute the SchemaChanges using field IDs */
+  static SchemaChanges computeSchemaChangesById(
```

Comment:
I wonder if we should have our tests use whatever method we expect to use to generate these `Map<Long, StructField> currentFieldIdToField` from a schema?